ci-operator/step-registry/cucushift/installer/rehearse/gcp/cases/valid-confidential-computing/provision/cucushift-installer-rehearse-gcp-cases-valid-confidential-computing-provision-commands.sh

```diff
@@ -0,0 +1,344 @@
+GOOGLE_PROJECT_ID="$(< ${CLUSTER_PROFILE_DIR}/openshift_gcp_project)"
+BASE_DOMAIN="$(< ${CLUSTER_PROFILE_DIR}/public_hosted_zone)"
+REGION=${LEASED_RESOURCE}
+
+CLUSTER_PREFIX="${NAMESPACE}-${UNIQUE_HASH}"
+
+ssh_pub_key=$(<"${CLUSTER_PROFILE_DIR}/ssh-publickey")
+pull_secret=$(<"${CLUSTER_PROFILE_DIR}/pull-secret")
+
+patch=$(mktemp)
+expected_err_msg=""
+ret=0
+
+declare -A cc_type_machine_series_dict=(
+  [AMDEncryptedVirtualization]="c2d, n2d, c3d"
+  [AMDEncryptedVirtualizationNestedPaging]="n2d"
+  [IntelTrustedDomainExtensions]="c3"
+  [Enabled]="c2d, n2d, c3d"
+)
+
+declare -A instance_type_supported_cc_type_dict=(
+  [n2d-standard-4]="AMDEncryptedVirtualization AMDEncryptedVirtualizationNestedPaging Enabled"
+  [c2d-standard-4]="AMDEncryptedVirtualization Enabled"
+  [c3d-standard-4]="AMDEncryptedVirtualization Enabled"
+  [c3-standard-4]="IntelTrustedDomainExtensions"
+  [n2-standard-4]=""
+)
+
+declare -A instance_type_unsupported_cc_type_dict=(
+  [n2d-standard-4]="IntelTrustedDomainExtensions"
+  [c2d-standard-4]="IntelTrustedDomainExtensions AMDEncryptedVirtualizationNestedPaging"
+  [c3d-standard-4]="IntelTrustedDomainExtensions AMDEncryptedVirtualizationNestedPaging"
+  [c3-standard-4]="AMDEncryptedVirtualization AMDEncryptedVirtualizationNestedPaging Enabled"
+  [n2-standard-4]="AMDEncryptedVirtualization AMDEncryptedVirtualizationNestedPaging Enabled IntelTrustedDomainExtensions"
+)
+
+function save_artifacts()
+{
+  local -r install_dir="$1"
+  local -r testing_scenario_num="$2"
+  local current_time
+
+  set +o errexit
+  current_time=$(date +%s)
+  sed '
+    s/password: .*/password: REDACTED/;
+    s/X-Auth-Token.*/X-Auth-Token REDACTED/;
+    s/UserData:.*,/UserData: REDACTED,/;
+    ' "${install_dir}/.openshift_install.log" > "${ARTIFACT_DIR}/cluster_${testing_scenario_num}_openshift_install-${current_time}.log"
+
+  set -o errexit
+}
+
+function create_install_config()
+{
+  local -r cluster_name=$1
+  local -r install_dir=$2
+
+  cat > ${install_dir}/install-config.yaml << EOF
+apiVersion: v1
+baseDomain: ${BASE_DOMAIN}
+compute:
+- architecture: amd64
+  hyperthreading: Enabled
+  name: worker
+  platform: {}
+  replicas: 3
+controlPlane:
+  architecture: amd64
+  hyperthreading: Enabled
+  name: master
+  platform: {}
+  replicas: 3
+metadata:
+  creationTimestamp: null
+  name: ${cluster_name}
+networking:
+  clusterNetwork:
+  - cidr: 10.128.0.0/14
+    hostPrefix: 23
+  machineNetwork:
+  - cidr: 10.0.0.0/16
+  serviceNetwork:
+  - 172.30.0.0/16
+platform:
+  gcp:
+    projectID: ${GOOGLE_PROJECT_ID}
+    region: ${REGION}
+publish: External
+pullSecret: >
+  ${pull_secret}
+sshKey: |
+  ${ssh_pub_key}
+EOF
+}
+
+function compose_expected_err_msg()
+{
+  local -r install_config_file="$1"
+  local instance_type on_host_maintenance1 on_host_maintenance2 cc_type1 cc_type2 supported_cc_types unsupported_cc_types
+
+  instance_type=$(yq-go r "${install_config_file}" compute[0].platform.gcp.type)
```

Comment:
Similar comment, we also can get the exact instance type for cp and computes nodes individually.